### DISCUSSION TITLE
SWT Supported Java Versions

### DISCUSSION THREAD

**Discussion by willynganga** [eclipse-platform/eclipse.platform.swt#1084] [General]
Is there any official documentation regarding the supported Java versions by SWT versions? This would be helpful when updating the dependency.

**Comment by mickaelistria** [comment]
The META-INF/MANIFEST.MF files in SWT jars tells the minimal version of Java it's compatible with.

**Reply by willynganga** [reply]
Hello @mickaelistria, thank you for your response.

I believe this is not an efficient way to get the information. I also noticed that the Build-Jdk-Spec  META-INF/MANIFEST.MF sometimes does not mean the compiled classes are targeting the Java version specified. For example, checking `org.eclipse.swt.gtk.linux.x86_64-3.120.0.jar` it says `Build-Jdk-Spec: 11` where's `org.eclipse.swt.gtk.linux.x86_64-3.121.0.jar` it says `Build-Jdk-Spec: 17` and if you use this with Java 11, it still compiles and works. This is a bit confusing, hence the need for official documentation.

**Reply by laeubi** [reply]
You need to look at `Bundle-RequiredExecutionEnvironment` header. If you like of course you can create such "official documentation" e.g. here: https://github.com/eclipse-platform/eclipse.platform.swt/blob/master/README.md
